build-docker.sh

```diff
@@ -1,27 +1,36 @@
 #!/usr/bin/env bash
-set -e
+set -e -o pipefail
 
-CONTAINER_NAME=trezor-firmware-env.nix
+cd "$(dirname "${BASH_SOURCE[0]}")"
+
+CONTAINER_NAME=${CONTAINER_NAME:-trezor-firmware-env.nix}
+ALPINE_VERSION=${ALPINE_VERSION:-3.12.0}
+ALPINE_RELEASE=${ALPINE_RELEASE:-3.12}
+ALPINE_ARCH=${ALPINE_ARCH:-x86_64}
+ALPINE_CDN=${ALPINE_CDN:-http://dl-cdn.alpinelinux.org/alpine}
+CONTAINER_FS_URL=${CONTAINER_FS_URL:-"$ALPINE_CDN/v$ALPINE_RELEASE/releases/$ALPINE_ARCH/alpine-minirootfs-$ALPINE_VERSION-$ALPINE_ARCH.tar.gz"}
 
 TAG=${1:-master}
 REPOSITORY=${2:-local}
 PRODUCTION=${PRODUCTION:-1}
 MEMORY_PROTECT=${MEMORY_PROTECT:-1}
 
-if [ "$REPOSITORY" = "local" ]; then
-  REPOSITORY=file:///local/
+if [[ "$REPOSITORY" = "local" ]]; then
+  REPOSITORY="file:///local/"
 else
-  REPOSITORY=https://github.com/$REPOSITORY/trezor-firmware.git
+  REPOSITORY="https://github.com/$REPOSITORY/trezor-firmware.git"
 fi
 
-wget -nc -P ci/ http://dl-cdn.alpinelinux.org/alpine/v3.12/releases/x86_64/alpine-minirootfs-3.12.0-x86_64.tar.gz
+wget --no-config -nc -P ci/ "$CONTAINER_FS_URL"
 docker build -t "$CONTAINER_NAME" ci/
 
-USER=$(ls -lnd . | awk '{ print $3 }')
-GROUP=$(ls -lnd . | awk '{ print $4 }')
+USER=$(id -u)
+GROUP=$(id -g)
```

Comment:
We want to keep the original behavior - this does something very different.

Reply:
Well, I touched it because original `ls -lnd` failed on macOS. There is no `-n` option available. Also output from `ls` may not be stable. E.g. because of some env variables or .rc files, mine contains headers at the top. I thought original intention was to detect effective user/group ids. Is there a better way to do this?

Reply:
The original intention is to keep the same user/group for newly created files as the root directory has. That means if the root dir is owned by stick/users and an user somebodyelse runs the script, the generated files will still have stick/users ownership.

Reply:
Hmm, `stat -c "%u:%g" .` might be a good replacement. But it has different interface under mac... `stat -f "%u:%g" .`

Reply:
Exactly. Does `stat --printf "%u:%g\n"` work on macOS?

Reply:
Maybe the following works?

```
USER=$(stat -c "%u" . || stat -f "%u" .)
GROUP=$(stat -c "%g" . || stat -f "%g" .)
```

Reply:
It does not. What about running it in nix-shell? Overkill?

Reply:
For me it worked, but produced extra output to stderr.

```
> USER=$(stat -c "%u" . || stat -f "%u" .)
stat: illegal option -- c
usage: stat [-FlLnqrsx] [-f format] [-t timefmt] [file ...]
> echo $USER
502
```

This works fine (no errors):
```
USER=$(stat -c "%u" . 2>/dev/null || stat -f "%u" .)
GROUP=$(stat -c "%g" . 2>/dev/null || stat -f "%g" .)
```

Reply:
Ah, bad idea. We have nix-shell inside the docker and it would use platform stat anyways. Well. I can branch the command based on detected OS.